src/lib/comms/include/sol-http.h

```diff
@@ -236,9 +250,108 @@ sol_http_param_init(struct sol_http_param *params)
 
 bool sol_http_param_add(struct sol_http_param *params,
     struct sol_http_param_value value) SOL_ATTR_WARN_UNUSED_RESULT;
+bool sol_http_param_add_copy(struct sol_http_param *params, struct sol_http_param_value value);
 void sol_http_param_free(struct sol_http_param *params);
 
-int sol_http_escape_string(char **escaped, const char *value);
+/**
+ * Encodes an URL string.
+ *
+ * If the string was not encoded this function will simple copy
+ * the slices content to the buffer and set @c SOL_BUFFER_FLAGS_MEMORY_NOT_OWNE.D
+ *
+ * @param buf The buffer that will hold the encoded string.
+ * @param value A slice to be encoded.
+ *
+ * @return 0 on success, negative number on error.
+ */
+int sol_http_encode_slice(struct sol_buffer *buf, const struct sol_str_slice value);
+
+/**
+ * Decodes an URL string.
+ *
+ * If the string was not decoded this function will simple copy
+ * the slices contents to the buffer and set @c SOL_BUFFER_FLAGS_MEMORY_NOT_OWNED.
+ *
+ * @param buf The buffer that will hold the decoded string.
+ * @param value A slice to be decoded.
+ *
+ * @return 0 on success, negative number on error.
+ */
+int sol_http_decode_slice(struct sol_buffer *buf, const struct sol_str_slice value);
+
+/**
+ *  Creates an URI based on struct sol_http_url and its params
+ *
+ * @param url_out The created URI - it should be freed using free().
+ * @param url The url parameters.
+ * @param params The query and cookies params.
+ *
+ * @return 0 on success, negative number on error.
+ */
+int sol_http_create_uri(char **uri_out, const struct sol_http_url url, const struct sol_http_param *params);
+
+/**
+ * A simpler version of sol_http_create_uri().
+ *
+ *
+ * @param uri The created URI - it should be freed using free().
+ * @param base_uri The base uri to be used.
+ * @param params The query and cookies params.
```

Comment:
`and cookies`? There is no meaning to have cookies in an URI :-D (same for the `sol_http_create_uri()`)